.github/workflows/end2end.yml

```diff
@@ -55,9 +55,13 @@ jobs:
           echo "CONTRACT_ADDRESS=$(forge script script/Deployment.s.sol:NovaVerifierDeployer --fork-url $ANVIL_URL --private-key $ANVIL_PRIVATE_KEY --broadcast --non-interactive | sed -n 's/.*Contract Address: //p' | tail -1)" >> $GITHUB_OUTPUT
         id: deployment
 
+      - name: Generate proof and public parameters from commit hash
+        run: |
+          python ci_pasta_keys_verifier_script.py https://github.com/lurk-lab/Nova.git d5f5fb5e557b99cb111f2d5693aa34fe30722750
```

Comment:
I wouldn't keep this information as a part of CI configuration. Instead, I would prefer explicitly having additional file in the repository with URL and commit hash (let's say `rust-reference-info.txt` or any other format convenient to utilise by CI). Then, during this step, CI should read that info and pass it as input to Python script.

Reply:
Agreed, but I think even better would be to keep them as repository variables, e.g. `$NOVA_URL` and `$NOVA_COMMIT`

Reply:
> Agreed, but I think even better would be to keep them as repository variables, e.g. $NOVA_URL and $NOVA_COMMIT

That simplifies fetching the information about reference in CI config, but we are losing: 1) some explicitness (as definition of those variables are hidden in repository settings) and 2) some flexibility if we want to target multiple branches (which is the case, as soon as we have `pasta`, `grumpkin` and I believe eventually we will need e2e testing in `gas_optimising` and `main`)